### PR TITLE
Backend/bug/tracing interceptor tests

### LINKED ISSUE #1336: Fix backend tests
Update in grpc library broke tests for our tracing interceptor

### PR DESCRIPTION
**Pull request by lennartvanlaake**
- [x] Formatted my code by running `isort . && black .` in `app/backend`
- [x ] Added tests for any new code or added a regression test if fixing a bug
- [x] All tests pass
- [ ] Run the backend locally and it works
- [x] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

### CODE REVIEW

**Comment by aapeliv** [comment]
They exposed the status code in the public API (though labelled experimental), so we no longer need to dig into the private state and play with CVs.

This should be enough:

```diff
diff --git a/app/backend/src/couchers/interceptors.py b/app/backend/src/couchers/interceptors.py
index b753ebe8..2669429e 100644
--- a/app/backend/src/couchers/interceptors.py
+++ b/app/backend/src/couchers/interceptors.py
@@ -144,9 +144,7 @@ class TracingInterceptor(grpc.ServerInterceptor):
             except Exception as e:
                 finished = perf_counter_ns()
                 duration = (finished - start) / 1e6  # ms
-                # need a funky condition variable here, just in case
-                with context._state.condition:
-                    code = context._state.code
+                code = context.code()
                 traceback = "".join(format_exception(type(e), e, e.__traceback__))
                 user_id = getattr(context, "user_id", None)
                 self._store_log(method, getattr(code, "name", None), duration, user_id, request, None, traceback)
@@ -175,9 +173,7 @@ class ErrorSanitizationInterceptor(grpc.ServerInterceptor):
             try:
                 res = prev_func(req, context)
             except Exception as e:
-                # need a funky condition variable here, just in case
-                with context._state.condition:
-                    code = context._state.code
+                code = context.code()
                 # the code is one of the RPC error codes if this was failed through abort(), otherwise it's None
                 if not code:
                     logger.exception(e)
```

File: app/backend/src/tests/test_interceptors.py [hunk 1]
```diff
@@ -218,7 +218,7 @@ def TestRpc(request, context):
 
 def test_tracing_interceptor_abort(db):
     def TestRpc(request, context):
-        context.abort(grpc.StatusCode.FAILED_PRECONDITION, "now a grpc abort")
+        context.abort(grpc.StatusCode.FAILED_PRECONDITION, "")
```
Comment (aapeliv) [inline app/backend/src/tests/test_interceptors.py]: This should now work again, I think?

Comment (lennartvanlaake) [reply app/backend/src/tests/test_interceptors.py]: yep, putting it back now

**Review by aapeliv — APPROVED** [review]
LGTM! Thanks :)

**Pull request merged**